File: tests/bin/socket.c
// SPDX-License-Identifier: Apache-2.0

#include "libc.h"
#include <sys/socket.h>

int main(void) {
    if (socket(AF_UNIX, SOCK_STREAM | SOCK_CLOEXEC, 0) < 0)
        return 1;

    if (socket(AF_INET, SOCK_STREAM | SOCK_CLOEXEC, 0) < 0)
        return 2;

    if (socket(AF_INET, SOCK_DGRAM | SOCK_CLOEXEC, 0) < 0)
        return 3;

    return 0;
}

File: tests/crates/enarx_syscall_tests/src/bin/socket.rs
// SPDX-License-Identifier: Apache-2.0

#![no_std]
#![no_main]
#![feature(naked_functions, asm_sym)]

enarx_syscall_tests::startup!();

use enarx_syscall_tests::*;

const AF_UNIX: i32 = 1;
const AF_INET: i32 = 2;
const SOCK_DGRAM: i32 = 2;

fn main() -> Result<()> {
    socket(AF_UNIX, libc::SOCK_STREAM | libc::SOCK_CLOEXEC, 0)?;
    socket(AF_INET, libc::SOCK_STREAM | libc::SOCK_CLOEXEC, 0)?;
    socket(AF_INET, SOCK_DGRAM | libc::SOCK_CLOEXEC, 0)?;
    Ok(())
}
